broadinstitute/single_cell_portal_core
# fixing error display when study is behind a download agreement (SCP-4006)

SCP-4006.  Yikes, our API error handling has a LOT of edge cases.  I've created SCP-4010 to handle a more substantive upgrade, and hopefully this is a step in the right direction.  there are two basic problems:
1.  on the frontend, we don't want/need custom handling for 401/403 errors on every UI component since they are almost always the user getting signed out.  (since the UX shouldn't be showing options the user isn't allowed to do)
2. The frontend can't rely on a single, canonical format of error response, since devise, rails, and our base API controller all report differently.

In any case, this updates the UX for if you try to do a bulk download including a study that needs a download agreement
![image](https://user-images.githubusercontent.com/2800795/150210732-2b52f7e6-71a2-4c19-95ee-2bf5220c9d74.png)

TO TEST:
1.  Add a download agreement to a study
2. sign in as a user who is not the study owner and who has not accepted the agreement
3. do a search which includes the study
4. click 'Download'
5. observe the modal above
6. Sign out
7. In `DownloadButton.js`, line 24, set `active = true` to enable the UX download even if you are not signed in
8. Reload the page, and click 'Download'
9. confirm a popup appears suggesting your session has expired
